README.md

```diff
@@ -1,2 +1,105 @@
 # elixir-base58-encode
 https://hex.pm/packages/b58
+
+
+This module provides the `Base58Encode.encode/1` function which takes an **Elixir binary** and returns its base58 String representation.
+
+See the section [What is an Elixir binary?](#What-is-an-elixir-binary) for more information about the binary type in Elixir.
+
+## How to use the module
+
+1. Add the package to you dependencies
+
+  ```
+  defp deps do
+    [b58: "~> 0.1.0"]
+  end
+  ```
+
+  and run `mix deps.get`
+
+2. Call the `encode` function with a binary as parameter:
+
+ ```
+ Base58Encode.encode("foo")
+ "bQbp"
+ ```
+
+ if the parameter is not an Elixir binary the function will return `:error`. In
+ this example 42 is an Integer and not a binary (`is_binary(42)` will return `false`)
+
+ ```
+ Base58Encode.encode(42)
+ :error
+ ```
+
+ See `example/Example.exs` file for a simple example on how to use the module.
+ You can also run this example with `mix run example/Example.exs`
```

Comment:
👍 

Really like this.

My one 'small' question is why does the `Example.exs` file start with a capitol letter? Not a blocker in any way. Just not normal convention is it? (think this might be elm syntax but I could be wrong)

Reply:
:smile: yes still some elm in my head. thanks for spotting that